compat/xis.h

```diff
@@ -14,6 +14,7 @@
 #endif
```

Comment:
AFAICT, we should include `<ctype>` in the C++ section below _and_ move these three lines inside the C-only section below. I know that we have been talking about these two basic changes for a while now, but I cannot tell why we cannot arrive at the final result that has `<ctype>` for C++ and these three lines for C.

Reply:
I had test-built it and it worked, so it might be that it's indirectly included. This said, it doesn't hurt and it clarifies intent so why not.
Test-building and adding it.

Reply:
> I had test-built it and it worked

FWIW, the latest published/reviewed version (i.e. commit [63d499c](https://github.com/squid-cache/squid/pull/1314/commits/63d499c8271e231b440af014bd8054fb2ef516cb)) worked because the file _is_ included (once, before C/C++ checks, just like it is included in the official code).